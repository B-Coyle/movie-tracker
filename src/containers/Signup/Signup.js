import React, { Component } from 'react';
import './Signup.scss';

class Signup extends Component {
    //extends signin? 
    constructor() {
        super()
        this.state ={
            name: '',
            email: '',
            password: '',
            status: '',
            validUser: false,


        }
    }
    handleChange = (e) =>{
        const {name, value} = e.target
        this.setState({
            [name]: value
        })
    }

    handleSubmit = (e) => {
        e.preventDefault();
        const {name} = e.target
        this.createUser();
        this.refs.name.value = '';
        this.refs.email.value = '';
        this.refs.password.value = '';
    }

    createUser = (state) => {
        const {name, email, password} = this.state
        const url = 'http://localhost:3000/api/users/new'
        const options = {
          method: 'POST',
          body: JSON.stringify({
            name,
            email,
            password
          }),
          headers: {
            'Content-Type': 'application/json'
          }
        }
        fetch(url, options)
        .catch(error => console.log(error));
        
      }

    render(){
        return(
            <div className='signup-container'>
            <header>
              <h1>Movie <i className="fas fa-film"></i> Tracker</h1>
            </header>
            <main>
            <form className='signup' onSubmit={this.handleSubmit}>
                <input
                value={this.state.name}
                name='name'
                placeholder="Your Name"
                onChange={this.handleChange}
                className='input'
                ref='name'
            />
                <input
                    value={this.state.email}
                    name='email'
                    placeholder="Email"
                    onChange={this.handleChange}
                    className="input"
                    ref='email'
                />
                <input
                    value={this.state.password}
                    type='password'
                    name='password'
                    placeholder="Password"
                    onChange={this.handleChange}
                    className="input"
                    ref='password'
                />
                <button onClick={this.addUser} className="create-account">Sign Up</button>
        </form>
        </main>
      </div>
        )
    }
}

export default Signup;