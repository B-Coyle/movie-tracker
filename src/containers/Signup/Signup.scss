.create-account {
    background-color: #307F9F;
    border: 0;
    color: #FFFFFF;
    font-size: 1.25em;
    font-weight: normal;
    margin: 1.5rem 6.5rem 1.5rem 0;
    outline: none;
    padding: .75rem 1rem;
    width: 250px;
    height: 50px;
    cursor: pointer;
}

main {
        display: flex;
        justify-content: center;
        margin: 0 auto;
}
      
main form {
        margin-top: 40px;
        display: flex;
        flex-direction: column;
        width: 250px;
}

header {
    padding-top: 50px;
        color: white;
        font-family: Anton;
}
      
h1 {
        margin: 0 auto 25px auto;
        font-size: 100px;
        color: #307F9F;
}