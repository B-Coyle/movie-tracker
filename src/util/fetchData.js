export const fetchData = async(url, options) => {
    const response = await fetch(url, options)
    if(response.ok) {
        return response.json()
    } else {
        throw Error('Error receiving data from fetch')
    }
}   

export const addUser = async (user) => {
    try  {
        const response = await fetch('http://localhost:3000/api/users/new', {
            method: 'POST',
            body: JSON.stringify(user),
            headers: {
            'Content-Type': 'application/json'
            }
        })
        const data = await response.json()
        return data
    } catch(error) {
        throw new Error()
    }
}
   