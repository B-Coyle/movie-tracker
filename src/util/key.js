export const key = "fb457ec4329812194b77f8e9822b2001";
