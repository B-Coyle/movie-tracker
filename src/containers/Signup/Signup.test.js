import React from 'react';
import ReactDOM from 'react-dom';
import Signup from './Signup';
import {shallow} from 'enzyme';
import {validateEmail} from './Signup';
import { addUser } from '../../util/fetchData';
import { Provider } from 'react-redux';
//npm i redux-mock-store
import configureMockStore from 'redux-mock-store'


jest.mock('../../util/fetchData')

const mockStore = configureMockStore();
const store = mockStore({});

beforeEach(() => {
  addUser.mockImplementation(() => ({error: false}))
})

describe('Signup', () => {
  let wrapper

  beforeEach(() => {
    wrapper = shallow(
    <Provider store={store}><Signup /></Provider>)
  })

it('should match the snapshot', () => {
  expect(wrapper).toMatchSnapshot();
  });

  it('should match the snapshot if it renders the redirect', () => {
    wrapper.state().validUser = true
    expect(wrapper).toMatchSnapshot()
  });

  it('should have a default state', () => {
    const expected ={
      name: '',
      email: '',
      password: '',
      status: '',
      validUser: false,
      formCompleted: false,
      validEmail: false,
      duplicateUser: false,
      error: ''
    }
    expect(wrapper.state()).toEqual(expected)
  })

  describe('validateEmail', () => {

    it('should return validEmail as true if no email is valid', () => {
      const email = wrapper.state().email = 'bac@gmail.com';
      const result= wrapper.instance().validateEmail(email);
      expect(result).toEqual(true)
    });

    it('should return validEmail as false if email is invalid', () => {
      const email = wrapper.state().email = 'bacgmail';
      const result = wrapper.instance().validateEmail(email);
      expect(result).toEqual(false);
    })

    describe('handleChange', () => {

      const mockEvent = {target: {name: 'Anon', value: 'anon@gmail.com'}}

      it.skip('should target and update the name in default state', () => {
        const expected = 'Anon'
        wrapper.instance().handleChange(mockEvent)
        expect(wrapper.state().name).toEqual(expected)
      });

    })

  })

})