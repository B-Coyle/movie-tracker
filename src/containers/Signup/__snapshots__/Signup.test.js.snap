// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Signup should match the snapshot 1`] = `
<div
  className="signup-container"
>
  <header>
    <h1>
      Movie 
      <i
        className="fas fa-film"
      />
       Tracker
    </h1>
  </header>
  <main>
    <form
      className="signup"
      onSubmit={[Function]}
    >
      <input
        className="input"
        name="name"
        onChange={[Function]}
        placeholder="Your Name"
        value=""
      />
      <input
        className="input"
        name="email"
        onChange={[Function]}
        placeholder="Email"
        value=""
      />
      <input
        className="input"
        name="password"
        onChange={[Function]}
        placeholder="Password"
        type="password"
        value=""
      />
      <h3
        className="email-in-use"
      />
      <button
        className="create-account"
      >
        Sign Up
      </button>
      <Link
        to="/"
      >
        <button
          className="return-login"
        >
          Return To Login
        </button>
      </Link>
    </form>
  </main>
</div>
`;

exports[`Signup should match the snapshot if it renders the redirect 1`] = `
<div
  className="signup-container"
>
  <header>
    <h1>
      Movie 
      <i
        className="fas fa-film"
      />
       Tracker
    </h1>
  </header>
  <main>
    <form
      className="signup"
      onSubmit={[Function]}
    >
      <input
        className="input"
        name="name"
        onChange={[Function]}
        placeholder="Your Name"
        value=""
      />
      <input
        className="input"
        name="email"
        onChange={[Function]}
        placeholder="Email"
        value=""
      />
      <input
        className="input"
        name="password"
        onChange={[Function]}
        placeholder="Password"
        type="password"
        value=""
      />
      <h3
        className="email-in-use"
      />
      <button
        className="create-account"
      >
        Sign Up
      </button>
      <Link
        to="/"
      >
        <button
          className="return-login"
        >
          Return To Login
        </button>
      </Link>
    </form>
  </main>
</div>
`;
